src/full_coverage_path_planner.cpp

```diff
@@ -69,9 +62,8 @@ namespace full_coverage_path_planner
   {
     geometry_msgs::msg::PoseStamped new_goal;
     std::list<Point_t>::const_iterator it, it_next, it_prev;
-    int dx_now, dy_now, dx_next, dy_next, move_dir_now = 0, move_dir_next = 0;
     bool do_publish = false;
-    float orientation = eDirNone;
+    float orientation = dir::none;
     RCLCPP_INFO(rclcpp::get_logger("FullCoveragePathPlanner"), "Received goalpoints with length: %lu", goalpoints.size());
     if (goalpoints.size() > 1)
```

Comment:
Can you elaborate more on your previous [comment](https://github.com/nobleo/full_coverage_path_planner/pull/25#discussion_r677243381)?

Reply:
Sure!

Early-outs make the code more readable by handling the exceptions in a short block at the top.

```
function do_awesome()
{
  if (everything_okay_as_expected)
  {
    Do 
    lots 
    of 
    indented stuff
  }
  else
  {
    warn about it
  }
}
```
Can be replaced by:

```
function do_awesome()
{
  if (!everything_okay_as_expected)
  { 
    warn about it
    return;
  }

  Do 
  lots 
  of 
  stuff
}
```

Reply:
And here it looks like `goalpoints.size() <= 1` is exceptional.

Reply:
Got it... in our case the function does not return right after the else condition, so we can not early-out. I can switch the order to at least show the especial case first.

Reply:
Aaah, I thought it did!
Then you may leave it like this as well. Up to you.

Reply:
On a similar note. I think the node dies if `goalpoints.size() == 0`. Since that case isn't handled.

Reply:
Yep.. will add that one